[Feature] Library file reading

Majkel-code opened the pull request:
.

Majkel-code committed:
new file

Majkel-code committed:
delete file

Majkel-code committed:
delete file "test" and create new file with correct name

Majkel-code committed:
Checking books.txt and users.txt if there isn't files create new empty.
If new user has the same nickname as some in database then return input to introduce new different nickname

Majkel-code committed:
Checking books.txt and users.txt if there isn't files create new empty.

If new user has the same nickname as some in database then return input to introduce new different nickname

Majkel-code committed:
1. Added 'while' to automated library.
2. Updating 'add_new_book' -> can add couple books one by one.
3. Updating 'register_user' -> user_name can't be the same as other one in database. user_password now must be input when new user try register in library, password can't be the same as other in database.

Majkel-code committed:
1. Created 'registry.txt'
2. On 'registry.txt' will be saved users and books what their borrow.
3. Add 'password' as required value to borrow books

Majkel-code committed:
1. couple changes in 'borrow' function, in books.txt now books are saved with 'Taken' after title ( example: book1Taken) -> this should help in function "add_new_book" to check 'if' statement.
2. "deposit" function should now work properly and users can deposit books they have borrowed earlier.
3. Changes in 'if' statement in "add_new_book" function.
4. "add_new_book" -> 'print' should now have more clarifies value
5. in 'while' loop is added new 'choice' to start 'deposit' function.

orokitax commented on Library.py:
```diff
@@ -0,0 +1,131 @@
+        self.registry = []
+
+
+    def setup(self):
+        if os.path.exists("C:/Users/micha/Documents/Python/PytonMentoring/books.txt") and os.path.exists("C:/Users/micha/Documents/Python/PytonMentoring/users.txt"):
+            print("Library successful loaded! ")
+            self.books = open("books.txt").read().split()
+            # PRINT TESTOWY!!!
+            print(self.books)
+            self.users = open("users.txt").read().split()
+            # PRINT TESTOWY!!!
+            print(self.users)
+            self.registry = open("registry.txt").read().split()
+            # PRINT TESTOWY!!!
+            print(self.registry)
+
+        else:
+            print("Missing some database, Creating new one... ")
+            open("books.txt", "a")
+            open("users.txt", "a")
+            open("registry.txt", "a")
+            self.setup()
```
Każdy plik osobno sprawdzaj o dodaj` os.getcwd()` zamiast lokalnej scieżki.

orokitax replied:
Używaj `with open` przy czytaniu z plików, i `close `przy tworzeniu ich

Majkel-code replied:
DONE

orokitax commented on Library.py:
```diff
@@ -0,0 +1,131 @@
+                with open("registry.txt", "w") as f:
+                    f.write(file_data)
+            print(f"{user} just deposit book: {book}")
+        else:
+            print(f"{user} do not have book: {book}")
```
To może również oznaczać złe hasło, dodaj taki case

Majkel-code replied:
DONE

orokitax commented on Library.py:
```diff
@@ -0,0 +1,131 @@
+
+
+    def add_new_book(self):
+        book_name = input("Input book title (CamelCase): ")
+        while book_name.lower() != "x":
```
Dodaj komunikat jak wyjśc z pętli dodawania książek

Majkel-code replied:
DONE

orokitax commented on Library.py:
```diff
@@ -0,0 +1,131 @@
+                with open("books.txt", "w") as f:
+                    f.write(file_data)
+            self.registry.remove(f"{user}={book}")
+            with open("registry.txt", "r") as f:
+                file_data = f.read()
+                file_data = file_data.replace(f"{user}={book}", "")
+                with open("registry.txt", "w") as f:
+                    f.write(file_data)
```
`with open `wewnątrz innego `with open` to zły pomysł, usuń wcięcie w 64 linii

Majkel-code replied:
DONE

orokitax commented on Library.py:
```diff
@@ -0,0 +1,131 @@
+            break
+
+
+lib = Library()
+path = "C:/Users/micha/Documents/Python/PytonMentoring"
+lib.setup()
```
`os.getcwd()`

Majkel-code replied:
DONE

Majkel-code committed:
1. update self.books and self.users -> now creating JSON file
2. update 'add_new_book' -> now working on JSON file and on dict.
2a. 'add_new_book' now expect book author and how much pages book has.
3. Update 'register_user' -> now working on JSON file and on dict (Username = KEY, User password = VALUE)

Majkel-code committed:
1. update self.registry -> now creating JSON file
2. update 'setup' -> checking in the work location is file registry.json if not create
3. 'borrow' function should now work in the way (operation on JSON files and dict)
4. 'deposit' function should now work in the way (operation on JSON files and dict)
5. added 'password_available' function to improve 'register_user' function.

Majkel-code committed:
1. Now is one file 'database.json' where is storage users, books and registry.
2. creating function 'available_books_to_borrow' -> this function shows what's books can be borrowed for user.
2a. 'borrow' function now should work correct which database file.
3. creating function 'available_books_to_deposit' -> this function shows what's books can be deposit for user.
3a. 'deposit' function now should work correct which database file.
4. added "user_log_in" function that check correct login and password
5. added "login_available" and "password_available" -> functions checking that login or password is available.
6. "register_user" function should now work with new database file.
7. added "can_be_add" function where is checking the book exists in library.
8. "add_new_book" function now should work with new database file. (This function should work only on 'admin' account.
9. modify "while" loop. Now user on the begining chose between login or register.

Majkel-code committed:
1. now can't borrow when library is empty
2. Can't deposit when user do not have any books.

orokitax commented on Library.py:
```diff
@@ -0,0 +1,161 @@
+import json
+class Library:
+
+    def __init__(self):
+        self.database = {"books": [], "users": [{"login": "admin", "password": "admin"}], "registry": []}
+
```
Użyj` self.setup(`) w` __init__` , dzięki temu nie musisz tego ręcznie robić tylko za kazdym razem automatycznie odpali się setup.

orokitax commented on Library.py:
```diff
@@ -0,0 +1,161 @@
+                pass
+
+    def borrow(self, user):
+        books = []
+        self.available_books_to_borrow(books)
+        if len(books) > 0:
```
`if self.available_books_to_borrow(books)` wystarczy

orokitax commented on Library.py:
```diff
@@ -0,0 +1,161 @@
+                        f.write(json.dumps(self.database))
+                    print(f"{user} just borrow book: {book}")
+        else:
+            print("Library is empty, 'admin' should deliver new books or wait maybe some user will return book :) ")
```
Fajne komunikaty

orokitax commented on Library.py:
```diff
@@ -0,0 +1,161 @@
+                pass
+
+    def deposit(self, user):
+        books = []
+        self.available_books_to_deposit(user, books)
+        if len(books) > 0:
```
tak samo jak w `borrow`

orokitax commented on Library.py:
```diff
@@ -0,0 +1,161 @@
+                return False
+        return True
+
+    def password_available(self, user_password):
+        for i in self.database["users"]:
+            if i["password"] == user_password:
+                return False
+        return True
```
`for user in self.databse["users"]` wyglądałoby lepiej, inaczej nie wiadomo co to jest` i`

orokitax commented on Library.py:
```diff
@@ -0,0 +1,161 @@
+    def register_user(self, user_name, user_password):
+        if self.login_available(user_name) and self.password_available(user_password):
+            self.database["users"].append({"login": user_name, "password": user_password})
+            with open("database.json", "w") as f:
+                f.write(json.dumps(self.database))
```
wielokrotnie powtarza się ta funkcja w kodzie, zamknij to w osobnej metodzie i wywołuj gdzie trzeba
